internal/gitinterface/log_test.go

```diff
@@ -102,4 +99,218 @@ func TestGetCommitsBetweenRange(t *testing.T) {
+	}
+
+	commitIDs = append(commitIDs, commitHash)
+
+	// creating two children from the first commit
+
+	// in the visual, these will be commit 2 and commit 3
+
+	children := createChildrenCommits(t, repo, treeHashes, commit.Hash, 2)
+
+	commitIDs = append(commitIDs, children...)
+
+	// creating a child for commit 2, which in the visual will be commit 4
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[3], []plumbing.Hash{children[0]}, fmt.Sprintf("Test commit %v", 4), testClock)
+
+	commitHash, err = WriteCommit(repo, commit)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, commitHash)
+
+	// creating a merge commit from the two children, which in the visual will be commit 5
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[4], children, fmt.Sprintf("Test commit %v", 5), testClock)
+
+	commitHash, err = WriteCommit(repo, commit)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, commitHash)
+
+	// creating a child for commit 3, which in the visual will be commit 6
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[5], []plumbing.Hash{children[1]}, fmt.Sprintf("Test commit %v", 6), testClock)
+
+	commitHash, err = WriteCommit(repo, commit)
+	if err != nil {
+		t.Fatal(err)
+	}
+	commitIDs = append(commitIDs, commitHash)
+
+	// Git tree with merge commit structure without its commit trees and its values:
+	//
+	//  commit 4       commit 5         commit 6
+	//    │              │  │              │
+	//    └─► commit 2 ◄─┘  └─► commit 3 ◄─┘
+	//            │              │
+	//            └─► commit 1 ◄─┘
+
+	allCommits, err := GetCommitsFromCommitIDs(commitIDs, repo)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Run("Test commit 1", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[0], plumbing.ZeroHash)
+		assert.Nil(t, err)
+		// commit 1 is the first commit, so it should be the only commit returned
+		expectedCommits := []*object.Commit{allCommits[0]}
+		if !reflect.DeepEqual(expectedCommits, commits) {
+			t.Errorf("Expected and actual commits do not match. Expected: %v, Got: %v", expectedCommits, commits)
+		}
+	})
+
+	t.Run("Test commit 2", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[1], plumbing.ZeroHash)
+		assert.Nil(t, err)
+		// commit 2 is the first child of commit 1, so only it and commit 1 should be returned
+		expectedCommits := []*object.Commit{allCommits[1], allCommits[0]}
+		if !reflect.DeepEqual(expectedCommits, commits) {
+			t.Errorf("Expected and actual commits do not match. Expected: %v, Got: %v", expectedCommits, commits)
+		}
+	})
+
+	t.Run("Test commit 3", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[2], plumbing.ZeroHash)
+		assert.Nil(t, err)
+		// commit 3 is the second child of commit 1, so only it and commit 1 should be returned
+		expectedCommits := []*object.Commit{allCommits[0], allCommits[2]}
+		if !reflect.DeepEqual(expectedCommits, commits) {
```

Comment:
any specific reason for not using `assert.Equal` here? I think this was originally using assert?

Reply:
Assert errors out for this test, at least in my editor because it cannot compare expected commits with the returned commits, since the commits have unexported fields, which assert cannot process.

Reply:
Can you paste the error here? We don't need to assert private fields, and we use assert with similar inputs in other tests.

Reply:
@adityasaky I just ran it again, with assert, and it worked. I am guessing my test were flakey. Changed the reflect.DeepEqual to assert.Equal